book/website/reproducible-research/reviewing/reviewing-challenges.md

```diff
@@ -0,0 +1,122 @@
+
+This is especially true early on in a project when a community is still being built up around it.
+Contributors who make poor submissions now, with help and encouragement, may turn into the driving force of a project in the future.
+Reviews should always be encouraging and especially in the early stages, it can be better to err on the side of accepting rather than alienate potential collaborators.
+
+## The weight of responsibility
+
+When performing a review it can be helpful to have an appreciation of who is ultimately responsible for the changes.
+Reviewers who feel overly responsible are liable to spend too much time and be too critical in their reviews.
+
+The author of the code is ultimately responsible for its quality, not the reviewer.
+This is the defacto position because in the majority of projects, if an error is found to trace back to a particular commit later in a project's life, it is invariably the author of the commit who will be first in line to fix it.
+
+This shouldn't be a question of blame, despite the fact `git blame` is your go-to command for finding out who write a piece of code.
+In practice, the person who wrote a piece of code tends to be the person with the most context, and who can most efficiently find a fix.
+
+## Impulsive reactions
+
+Some review systems will send an email or alert every time a new comment is added to a review.
+It can be tempting, as either an author or reviewer, to respond to these comments immediately.
+This urge can be especially strong if you disagree with a comment.
+
+In practice the comment is likely to be part of a larger and more considered response that spans the entire set of changes in a pull request.
+
+Whether as a reviewer responding to changes, or an author responding review feedback, it can be helpful to wait for the entire set of comments to arrive before deciding on a response.
+Replying immediately not only starves you of the full context, it may also interrupt the flow of the reviewer, who may also be receiving immediate notifications following a reply.
+
+Resist the urge to reply to all comments immediately.
+Wait for the full response and then think carefully about how best to respond.
+
+The GitHub reviewing tools encourage this approach by allowing you to build up all of your comments in your review before submitting them in a single batch.
+This has many advantages:
+
+1. The recipient receives a single notification for the entire review, rather than many smaller notifications, which can be disruptive.
+2. The relevance of changes earlier in a pull request may not become clear until later in a pull request.
+   Batching up your responses gives you the opportunity to amend them in retrospect in case you discover that a comment doesn't apply after all.
+3. The approach encourages you to think holistically about a change, rather than pedantically criticising every small and insignificant detail.
+4. It discourages overly-hasty and impulsive reactions.
+
+
```

Comment:
i really like this section. I think I would at a section about frustration.
people are frustrated either because the change they made was annoying or the review was hard.
but one should remember to be respectful with the effort of others. When opening a pr i should take proper measures to make the reviewers job pleasent and when getting the review, remember that the reviwer spent time and effort and i have the opertunity to benefit from it.

on the other hand the reviwer needs to remember that the code was the result of (probably) hard work and that a rejected review comment may have better reasons than to annoy the reviewer